pkg/controller/nebulascheduledbackup/nebula_scheduled_backup_manager.go

```diff
@@ -0,0 +1,250 @@
+		if err != nil {
+			return fmt.Errorf("getting running nebula backup for Nebula scheduled backup %s/%s failed. err: %w", scheduledBackup.Namespace, scheduledBackup.Name, err)
+		}
+
+		// Detect failed nebula backup
+		for _, condition := range nebulaBackup.Status.Conditions {
+			if (condition.Type == v1alpha1.BackupFailed || condition.Type == v1alpha1.BackupInvalid) && condition.Status == corev1.ConditionTrue {
+				if err := bm.clientSet.NebulaScheduledBackup().UpdateNebulaScheduledBackupStatus(scheduledBackup, &kube.ScheduledBackupUpdateStatus{
+					Phase: v1alpha1.ScheduledBackupJobFailed,
+				}); err != nil {
+					return fmt.Errorf("update nebula scheduled backup %s/%s status err: %w", scheduledBackup.Namespace, scheduledBackup.Name, err)
+				}
+				return utilerrors.ReconcileErrorf("nebula backup for scheduled backup [%s/%s] failed to complete. reason: %s message: %s", scheduledBackup.Namespace, scheduledBackup.Name, condition.Reason, condition.Message)
+			}
+		}
+
+		// Detect complete nebula backup
+		if condutil.IsBackupComplete(nebulaBackup) {
+			err := bm.clientSet.NebulaBackup().DeleteNebulaBackup(nebulaBackup.Namespace, nebulaBackup.Name)
+			if err != nil {
+				return fmt.Errorf("delete nebula backup %s/%s err: %w", nebulaBackup.Namespace, nebulaBackup.Name, err)
+			}
+
+			if err = bm.clientSet.NebulaScheduledBackup().UpdateNebulaScheduledBackupStatus(scheduledBackup, &kube.ScheduledBackupUpdateStatus{
+				Phase: v1alpha1.ScheduledBackupScheduled,
+			}); err != nil {
+				return fmt.Errorf("update nebula scheduled backup %s/%s status err: %w", scheduledBackup.Namespace, scheduledBackup.Name, err)
+			}
+			return utilerrors.ReconcileErrorf("nebula backup for scheduled backup [%s/%s] complete succesfully.", scheduledBackup.Namespace, scheduledBackup.Name)
+		}
+		return utilerrors.ReconcileErrorf("waiting for nebula backup [%s/%s] for scheduled backup [%s/%s] to complete.", nebulaBackup.Namespace, nebulaBackup.Name, scheduledBackup.Namespace, scheduledBackup.Name)
+	} else {
+		currTime := metav1.Now()
+		if currTime.Equal(scheduledBackup.Status.NextBackupTime) || currTime.After(scheduledBackup.Status.NextBackupTime.Time) {
+			klog.Infof("next scheduled backup is up for Nebula scheduled backup %v/%v. Triggering Nebula backup job.", scheduledBackup.Namespace, scheduledBackup.Name)
+			lastBackupTime := scheduledBackup.Status.NextBackupTime
+			nextBackupTime, err := calculateNextRunTime(scheduledBackup, nil)
+			if err != nil {
+				return fmt.Errorf("calculating next runtime for nebula scheduled backup %s/%s failed. err: %w", scheduledBackup.Namespace, scheduledBackup.Name, err)
+			}
+
+			scheduledBackup.Spec.BackupTemplate.MaxBackups = scheduledBackup.Spec.MaxBackups
+			scheduledBackup.Spec.BackupTemplate.MaxReservedTime = scheduledBackup.Spec.MaxReservedTime
+			nebulaBackup := v1alpha1.NebulaBackup{
```

Comment:
Here OwnerReference is lost

Reply:
@MegaByte875 Got it, will add a owner reference when creating this.

Reply:
Fixed